GPIO used

PierreGdrlt opened the pull request:
I want to connect my scale on GPIO 3 (DT) and GPIO 27 (SCK). How can change for use this GPIO instead of 20 and 21 ???

techman83 committed:
Update for python3 + octoprint 1.5.3

techman83 committed:
Replace 'long' with 'int'

techman83 committed:
discard bad/outlier reads

techman83 committed:
Python 2.7 no longer supported

techman83 committed:
Line endings, whitespace, and tabs > spaces (PEP8)

techman83 committed:
Fix linting errors and add initial test

techman83 committed:
Clean up tests and add action workflow

techman83 committed:
Bump plugin version due to backwards compatibility not being guaranteed

techman83 committed:
Add octoprint to testing workflow

techman83 committed:
Update Readme

techman83 committed:
Add mock.gpio dependency

techman83 committed:
Revert plugin name for now

techman83 committed:
Merge pull request #1 from techman83/op153_refactor

Op153 refactor

techman83 committed:
Install required deps for package testing

techman83 committed:
Rename project

techman83 committed:
Update repo

techman83 committed:
Use a version file

techman83 committed:
chore: Add license file

techman83 committed:
chore: Add release process

Add's workflows for release building

techman83 committed:
Merge pull request #2 from techman83/release

Update Project and Add Release workflow

techman83 committed:
fix: Use mv instead of rename in release action

techman83 committed:
chore: Release to publish automatically

techman83 committed:
chore: Bump version and fix author/repo information

techman83 committed:
fix: use existing identifier for automatic change between plugins

techman83 committed:
fix: Ignore line length in update information

techman83 committed:
Merge pull request #3 from techman83/fix_views

Fix Views + Versioning

techman83 committed:
Merge pull request #4 from techman83/development

Release 0.2.2

techman83 committed:
chore: optimise release process

techman83 committed:
Merge pull request #5 from techman83/optimise_release_process

Optimise Release Process

techman83 committed:
Merge pull request #6 from techman83/development

Release v0.2.3

techman83 committed:
chore: release zip file

Make for easier linking from OctoPrint Plugin Registry

techman83 committed:
Merge pull request #7 from techman83/optimise_release_process

Update Release Process

techman83 committed:
chore: optimise release process

techman83 committed:
chore: Update latest asset link

techman83 committed:
chore: bump release version

techman83 committed:
chore: separate tagging from release for better release notes

techman83 committed:
chore: bump release version

techman83 committed:
chore: simplify release process

After various gyrations I settled on this process for now.

techman83 committed:
Merge pull request #8 from techman83/improve_releases

Improve Releases

techman83 committed:
fix: Long not connected message

The 'Sensor Not Connected' message was both long and slightly inaccurate. Which
lead to the interface jumping around whenever a string of bad readings came
through. Shortened the error to 'NaN' and documented it in the README.

techman83 committed:
Merge pull request #9 from techman83/shorten_error_message

Long Message in UI

techman83 committed:
test: Add test for dependency errors during installation

Reported in #10, installation failed as RPi.GPIO was missing during
setup.

techman83 committed:
fix: Dependency errors during setup

Open version file directly to avoid requiring runtime dependencies during setup - fixes #10

techman83 committed:
Merge pull request #11 from techman83/fix_install

Fix Setup

ali1234 committed:
Fix typo in default settings

This typo causes the plugin to fail for new installations. If you
have already configured it, these values are not used.

Fixes #12

techman83 committed:
Merge pull request #14 from ali1234/main

Fix typo in default settings

techman83 committed:
test: Ensure setting defaults are present

During refactoring a typo lead to default settings not being correct. Ensure our future defaults don't change without intent.

Relates to #12

techman83 committed:
Merge pull request #15 from techman83/defaults_test

Defaults Test

PierreGdrlt commented:
I've changed the 3 lines in __Init__.py :
36    clockpin=27
37    datapin=3
50    self.hx = HX711(3,27)

and the plugin always give error  "Error: No filament sensor detected"
I've restart Octoprint, and Pi.

anything else ?

parsko41 commented:
Glad you made progress!  There is a calibration process, if I recall.  It's been a while since I've used mine, as I have been going through printer changes.  There is a chance pins are backwards, or a connection issue exists.

-Parsko

PierreGdrlt commented:
I checked that my sensor was working fine. I put an SD card with Raspbian in my Pi, and with a proven program (the one that runs another scale on another Pi), I was able to make sure my sensor was functional, and properly plugged in. .
So I corrected the init.py file, and restart my Pi.
The result is always the same. I have no reading in the settings of FSE.
Where should I look for ??
![Snapshot_1](https://user-images.githubusercontent.com/82917872/117577956-12c15d80-b0ba-11eb-8da2-8908e9b764e2.png)

parsko41 commented:
Hi,
It appears that someone is upgrading this.  Can you have a look at the following link and see if that solves your problem?  I've reviewed [my own version of this](https://github.com/dieki-n/Octoprint-Filament-Scale/issues/13) where I changed my pin assignments, and I did not make any other changes  than you have.  Maybe you could also look at that.  You may have Python 3 issues perhaps?  I'm sorry I can't be further assistance.

https://github.com/OctoPrint/plugins.octoprint.org/pull/864

-Parsko

PierreGdrlt commented:
Well.
Here is the latest bad news. I have tried everything. I have re-installed Octoprint at least 4 or 5 times, a few times it works, and after a few minutes it doesn't work, and sometimes it doesn't work at all.
I give up ... I will do more tests maybe in a few months, if the plugin stabilizes, or maybe other things will be released in the meantime.

Good continuation.

parsko41 commented:
My experience with the HX711 chips is dodgy.  They do not provide stable reading, simply put.  I use them professionally, and can get them to work because I am writing the software.  But, even then, they give me sporatic readings that are simply wrong.  Using them with a Raspberry Pi is worse than a proper microcontroller.  It's a pseudo I2C protocol.  It's I2C, but does not work with I2C libraries, in my experience.  I have not tried again since I've gained much more experience with it.  I also haven't read the datasheet to get the exact timing between edges and such.  I'm sure there is one to get it to read reliably, perfectly.

A proper way to do this, IMHO, is to have a sub micro, like a RP2040 QTPY do the reading and filtering, while also maintaining the communication with the Pi.  Much cleaner and more reliable.  It's on my LOOONG list of things to do in the next couple years.

If it's worth anything, mine isn't hooked up right now.  I would really love this integrated with the spool manager.  Another item on the list....

Good luck, stay safe.

jcassel committed:
feat: MQTT Support

Building on @jcassel's work in #40, this publishes scale changes
to mqtt via the Octoprint-MQTT plugin.

closes #39

techman83 committed:
fix: Typing / Linting Errors

The code base was ported from python2 on a whim, and a lot of the code
reflects that. This tidies up the typing and linting a little more

techman83 committed:
fix: Read Average

Although the effort was done to create an average with outlier dection, it was
never actually used. This switches the function call over

techman83 committed:
ci: Add Commit Lint

The automatic releases use commit messages to generate the release notes. This
ensures the commits are correctly formmated to align with it.

techman83 committed:
ci: Bump Actions Versions

Actions Checkout + Actions Python have had a bunch of improvements, time to update
them.

techman83 committed:
Merge pull request #41 from techman83/feat/mqtt

Add MQTT Support
